bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -194,20 +204,30 @@ public class IpAddonFinder extends BaseAddonFinder {
     private static final String REPLACEMENT_UUID = "uuid";
 
     private final Logger logger = LoggerFactory.getLogger(IpAddonFinder.class);
+    private final NetworkAddressService networkAddressService;
     private final ScheduledExecutorService scheduler = ThreadPoolManager
             .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
     private final Set<AddonService> addonServices = new CopyOnWriteArraySet<>();
     private @Nullable Future<?> scanJob = null;
     Set<AddonInfo> suggestions = new HashSet<>();
 
-    public IpAddonFinder() {
+    @Activate
+    public IpAddonFinder(final @Reference NetworkAddressService networkAddressService) {
         logger.trace("IpAddonFinder::IpAddonFinder");
         // start of scan will be triggered by setAddonCandidates to ensure addonCandidates are available
+        this.networkAddressService = networkAddressService;
+    }
+
+    @Activate
+    public void activate() {
+        logger.trace("IpAddonFinder::activate");
+        networkAddressService.addNetworkAddressChangeListener(this);
```

Comment:
Is there a reason not to include this in the constructor?

Reply:
No, fixed.